Functioning connection failure detection

Geraden07 opened the pull request:
Will not run tests and repeatedly attempt to connect if initial connection test fails.

Geraden07 committed:
Functioning connection failure detection. Will not run tests and repeatedly attempt to connect if initial connection test fails.

Geraden07 commented:
Callback to Issue #13 .

sysgen reviewed (changes requested):
Changes ok however assertion failure message is too 'in the code' and should contain something that is understandable to the runner of the test (eg a non java dev, just a wM user) such as a message suggesting to check the host, port and credentials.

Geraden07 committed:
Refactored some Assert.fails as technical exceptions.

Geraden07 commented:
Hmm, interesting. I see what you're saying.

Here's the thing. In the event of an authentication failure, the message shown will actually be the 
```Unable to connect to <host> : <port> with user <username> - <exception message>"``` which comes from the ExecutionContext class connectToServer method. This AssertionError exception is thrown and bubbles up without being caught so that's what the user sees.

The other two messages I have coded are:
1. ``` Assert.fail("In testing connection to the server, isConnected came back false");```

This should never happen. Either testConnection returns true or throws the AssertionError with the "unable to connect...." message mentioned above.

I'm not a fan of having a testConnection method return void, just out of principle I suppose, but that might make things cleaner.

Or I could change it to an IllegalStateException instead of an Assertion fail, as I think it's more of a technical failure case. In that situation, the more "in the code" message is a little more appropriate. Do you agree?

2. ``` fail("Execution context is null");```

I figure this is a technical error situation and removed the check. I would prefer to still check for Null and throw the NPE myself with a message saying exactly WHAT is null, but the Codacy Quality Review tool was failing me with "Avoid throwing Null Pointer Exceptions", so I just removed it all together. This object should always be initialized at this point and Java is great at throwing NPEs all on its own if not.
